### PR TITLE
Docker Jupyter Image

### PR DESCRIPTION
**Pull request by mcarmonaa**
#### This PR must be merged after https://github.com/src-d/spark-api/pull/78

Changes:
- Added rules to `Makefile` to run and build our jupyter-spark-api container and bblfsh easily.
- Updated notebook example.
- Notebooks examples are now under `examples/notebooks`.
- Added siva-files for use from examples under `examples/siva-files`. They are some little repos that contain python and java files, so UAST extraction can be used.
- `bblfsh` endpoint configuration is loaded in the container from the `spark-defaults.conf`file. Spark load from this file those `keys` which start by `spark.` so our `tech.sourced.whatever` variable has been renamed to `spark.tech.sourced.whatever`.
- The `bblfsh scala client` version has been updated to `v0.2.0` to be compatible with last `bblfsh` versions.
- Updated `README` to show how to run jupyter and bblfsh containers.

### CODE REVIEW

**Comment by ajnavarro** [comment]
rebase please.

**Comment by mcarmonaa** [comment]
@ajnavarro rebase accomplished

File: Makefile [hunk 1]
```diff
@@ -3,24 +3,21 @@ DOCKER_CMD = docker
...
-# Scala version
-SCALA_VERSION ?= 2.11.11
+DOCKER_RMI = $(DOCKER_CMD) rmi -f
+
+# Docker run bblfsh server container
+BBLFSH_CONTAINER_NAME = bblfsh
+BBLFSH_HOST_PORT = 9432
+BBLFSH_CONTAINER_PORT = 9432
+BBLFSH_IMAGE = bblfsh/server
+BBLFSH_VERSION = latest
```
Comment (ajnavarro) [inline Makefile]: can you fix here the bblfsh version that is working with this spark-api version please?

File: README.md [hunk 2]
```diff
@@ -23,6 +23,9 @@ $ ./bin/spark-shell --packages "com.github.src-d:spark-api:master-SNAPSHOT" --re
 $ ./bin/pyspark --repositories "https://jitpack.io"  --packages "com.github.src-d:spark-api:master-SNAPSHOT"
 ```
 
+Run [bblfsh server](https://github.com/bblfsh/server):
+
+    docker run --rm --privileged -p 9432:9432 --name bblfsh bblfsh/server bblfsh server --log-level="debug"
```
Comment (ajnavarro) [inline README.md]: You can add also here the `make docker-bblfsh` command to do the same

File: README.md [hunk 3]
```diff
@@ -53,6 +56,14 @@ $ export SPARK_HOME=$HOME/spark-2.2.0-bin-hadoop2.7
 $ $SPARK_HOME/bin/spark-shell
 ```
 
+## bblfsh
+
+If you want to be able to use the UAST extraction features spark-api provides, you must run a Run [bblfsh server](https://github.com/bblfsh/server). You can do it easily with docker
+
+    docker run --rm --privileged -p 9432:9432 --name bblfsh bblfsh/server bblfsh server --log-level="debug"
```
Comment (ajnavarro) [inline README.md]: This is the same documentation that we have above, right?

File: Makefile [hunk 4]
```diff
@@ -75,11 +94,17 @@ build:
...
 	$(SBT) scalastyle test test:scalastyle coverage coverageReport
 
+docker-bblfsh:
+	$(DOCKER_RUN) $(BBLFSH_RUN_FLAGS)
+
 docker-build: build
-	$(DOCKER_BUILD) -t $(call unescape_docker_tag,$(DOCKER_IMAGE_VERSIONED)) .
+	$(DOCKER_BUILD) -t $(call unescape_docker_tag,$(JUPYTER_IMAGE_VERSIONED)) .
 
 docker-run: docker-build
```
Comment (ajnavarro) [inline Makefile]: to run the docker image, you don't need to build it before

File: README.md [hunk 5]
```diff
@@ -153,9 +164,13 @@ scala> api.getRepositories.filter('id === "github.com/mawag/faq-xiyoulinux").
 
 You can launch our docker container which contains some Notebooks examples just running:
 
-    docker run --name spark-api-jupyter -it --rm -p 8888:8888 -v $(pwd)/path/siva-files:/repositories src-d/spark-api-jupyter
+    docker run --name spark-api-jupyter --rm -it -p 8888:8888 -v $(pwd)/path/to/siva-files:/repositories --link bblfsh:bblfsh src-d/spark-api-jupyter
+
```
Comment (ajnavarro) [inline README.md]: we can add here also the `make docker-run` command

File: README.md [hunk 6]
```diff
@@ -179,25 +194,35 @@ $ make docker-build
 
 Notebooks under examples folder will be included on the image.
 
-To build and run a container with the Jupyter server
+To build and run a container with the Jupyter server:
```
Comment (ajnavarro) [inline README.md]: we don't need to build before run

File: README.md [hunk 7]
```diff
@@ -179,25 +194,35 @@ $ make docker-build
...
 
 ```bash
 $ make docker-run
 ```
 
-Container's output will show you an URL that you can paste on your browser.
+Before run the jupyter container you must run a bblfsh server:
+
+```bash
+$ make docker-bblfsh
```
Comment (ajnavarro) [inline README.md]: all this is duplicated, we should have the ways to run the spark-api docker container in the same place.

File: README.md (hunk 2 above)
Comment (mcarmonaa) [reply README.md]: In the quickstart section it supposed that you want to use spark-api in your project, no clone our repo, so you wouldn't have the Makefile. I'm wrong in the point of view?

File: README.md (hunk 3 above)
Comment (mcarmonaa) [reply README.md]: Yes it is, but they are in different section. Quickstart section just has commands to copy, and here there are explanations about why you need this things.

File: README.md (hunk 5 above)
Comment (mcarmonaa) [reply README.md]: Same than before, I supposed that they are instructions to use the image without clone the project. Should I change it?

File: README.md (hunk 7 above)
Comment (mcarmonaa) [reply README.md]: These instructions are instructions for a development point of view, when you are working on the spark-api.

**Comment by mcarmonaa** [comment]
Note that the `bblfsh` version used (`v.1.2.0`) works because `client-scala` dependency is updated to `v0.2.0` which is not a "stable" version. We should change the `client-scala` version to `v.1.0.0` when it is ready.

**Pull request merged**